lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/ShelleySpec.hs

```diff
@@ -199,17 +220,21 @@ instance {-# OVERLAPS #-} Arbitrary (Passphrase "seed") where
         bytes <- BS.pack <$> vectorOf n arbitrary
         return $ Passphrase $ BA.convert bytes
 
-newtype SingleAddress = SingleAddress Address deriving (Eq, Show)
+newtype SingleAddress (n :: NetworkDiscriminant)
+    = SingleAddress Address
+    deriving (Eq, Show)
 
-instance Arbitrary SingleAddress where
+instance KnownNetwork n => Arbitrary (SingleAddress n) where
     arbitrary = SingleAddress . Address . BS.pack
-        <$> vectorOf addrSingleSize arbitrary
+        <$> fmap ([single @n] <>) (vectorOf publicKeySize arbitrary)
 
-newtype GroupedAddress = GroupedAddress Address deriving (Eq, Show)
+newtype GroupedAddress (n :: NetworkDiscriminant)
+    = GroupedAddress Address
+    deriving (Eq, Show)
 
-instance Arbitrary GroupedAddress where
+instance KnownNetwork n => Arbitrary (GroupedAddress n) where
     arbitrary = GroupedAddress . Address . BS.pack
-        <$> vectorOf addrGroupedSize arbitrary
+        <$> fmap ([grouped @n] <>) (vectorOf (2*publicKeySize) arbitrary)
```

Comment:
in corresponding src module we have : 
```
publicKeySize :: Int
publicKeySize = 32

-- Serialized length in bytes of a Single Address
addrSingleSize :: Int
addrSingleSize = 1 + publicKeySize

-- Serialized length in bytes of a Grouped Address
addrGroupedSize :: Int
addrGroupedSize = addrSingleSize + publicKeySize
```
so it is `2*publicKeySize + 1` or I am missing something?

Reply:
It is, but the +1 is actually the `grouped` byte that is appended :)

Reply:
Right, I see now. For Arbitrary instance `SingleAddress` there is `publicKeySize + 1` (because of `single` size).